api/controllers/FileDownloadController.js

```diff
@@ -0,0 +1,66 @@
+const sails = require('sails');
+const request = require('request');
+const crypto = require('crypto');
+
+const FileDownloadController = {
+    downloadFileHandler(req, res) {
+        try {
+            const apiOptions = FileDownloadController._prepareAPIOptions(req);
+            FileDownloadController._streamFileToClient(apiOptions, res);
+        } catch (err) {
+            sails.log.error(err);
+            return res.serverError();
+        }
+    },
+
+    _prepareAPIOptions(req) {
+        if (req.params.apostilleRef === 'undefined') {
+            throw new Error('Missing apostille reference');
+        }
+
+        const {
+            hmacKey,
+            customURLs,
+            casebookCertificate: cert,
+            casebookKey: key,
+        } = req._sails.config;
+        const queryParamsObj = {
+            timestamp: new Date(Date.now()).getTime().toString(),
```

Comment:
you don't have to pass in `Date.now()` to `new Date(..)`.  `new Date()` (with no params) will return you a Date object using the date time that it was called. 

`Date.now()` will return you epoch time, so it looks like you just need `Date.now().toString()` (or use a template string) 

repl below - they return the same value 
````
> console.log(new Date(Date.now()).getTime().toString(), '\n', new Date().getTime().toString(), '\n', Date.now().toString(), '\n', `${Date.now()}`)
1632936724048 
 1632936724048 
 1632936724048 
 1632936724048
````